sdk/servicebus/service-bus/src/core/streamingReceiver.ts

```diff
@@ -402,9 +402,78 @@ export class StreamingReceiver extends MessageReceiver {
     await this._receiverHelper.suspend();
   }
 
-  async init(useNewName: boolean, abortSignal?: AbortSignalLike): Promise<void> {
-    const options = this._createReceiverOptions(useNewName, this._getHandlers());
-    await this._init(options, abortSignal);
+  private static wrapRetryOperation(
+    entityPath: string,
+    fullyQualifiedNamespace: string,
+    fn: () => Promise<void>,
+    onError?: OnError
+  ) {
+    return async () => {
+      try {
+        await fn();
+      } catch (err) {
+        onError &&
+          onError({
```

Comment:
This would mean that we would calling the user's error handler for every error that occurs. For retryable errors, previously, we would do this only if the retry attempts had exhausted. Are we sure that the increased noise is what we want?

Reply:
Now that we are reporting the error to the user's callback inside the while loop, we shouldnt need to do the same here right?

Reply:
On second thoughts, I dont see the need for `wrapRetryOperation` either...

Reply:
It's pretty minimal - it just handles tacking on the logic for wrapping most errors in a StreamingReceiverError vs just throwing directly (AbortError).